### PR TITLE
is817/fix-project schemas

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- [x]  [test_repo_data.py](api/tests/test_repo_data.py) validates some data files in repo against schemas
- [x] [Makefile](api/specs/webserver/v0/components/schemas/Makefile) to ease sync between json-schema and openapi.
- [x] Review project and workbench **schemas** (by @odeimaiz )
- [x] Review projects and workbench **data** (by @odeimaiz )

## Related issue number
#817 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

**Comment by pcrespov** [comment]
@odeimaiz, @sanderegg 
Please check the test results from [api/tests/test_repo_data.py](api/tests/test_repo_data.py) and could you please review (update & fix) the data and schemas ? Please commit all changes to this branch.

File: api/specs/webserver/v0/components/schemas/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# target: clean – Cleans all *-converted.yaml
+clean: check
+	- rm $(outputs)
+
+# target: help – Display all callable targets
+help:
+	@echo
+	@egrep "^\s*#\s*target\s*:\s*" [Mm]akefile \
+	| sed -r "s/^\s*#\s*target\s*:\s*//g"
+	@echo
```
Comment (sanderegg) [inline api/specs/webserver/v0/components/schemas/Makefile]: nice!!

File: api/specs/webserver/v0/components/schemas/project-v0.0.1-converted.yaml [hunk 2]
```diff
@@ -51,18 +51,19 @@ properties:
           enum:
```
Comment (sanderegg) [inline api/specs/webserver/v0/components/schemas/project-v0.0.1-converted.yaml]: the whole thing being under x-patternProperties, everything below that is not used. by the openapi validator..

File: api/tests/test_repo_data.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        - api/specs/webserver/v0/components/schemas/project-v0.0.1.json
+    """
+    data = _load_data(this_repo_root_dir / data_path)
+
+    assert any(isinstance(data, _type) for _type in [List, Dict])
+    if isinstance(data, Dict):
+        data = [data,]
+
+    for project_data in data:
+        jsonschema.validate(project_data["workbench"], workbench_schema)
```
Comment (sanderegg) [inline api/tests/test_repo_data.py]: cool this will help a lot!

File: api/specs/webserver/v0/components/schemas/project-v0.0.1.json [hunk 4]
```diff
@@ -97,20 +72,19 @@
...
             "position"
           ],
           "properties": {
-            "label": {
-              "type": "string"
-            },
             "key": {
               "type": "string",
               "description": "distinctive name for the node based on the docker registry path",
-              "pattern": "^(simcore)/(services)(/demodec)?/(comp|dynamic)(/[^\\s/]+)+$",
```
Comment (sanderegg) [inline api/specs/webserver/v0/components/schemas/project-v0.0.1.json]: can't we have a pattern that works for the registry key and at the same time for the others? now it's fully free... can we have either simcore/service/dyn|comp or a name such as group, file-picker?

Comment (odeimaiz) [reply api/specs/webserver/v0/components/schemas/project-v0.0.1.json]: Resolved in https://github.com/pcrespov/osparc-simcore/pull/21

Comment (sanderegg) [reply api/specs/webserver/v0/components/schemas/project-v0.0.1.json]: and I guess /demodec should be removed?

File: api/specs/webserver/v0/components/schemas/project-v0.0.1.json [hunk 5]
```diff
@@ -81,10 +81,11 @@
             "key": {
               "type": "string",
               "description": "distinctive name for the node based on the docker registry path",
+              "pattern": "^(simcore)/(services)(/demodec)?/(comp|dynamic|frontend)(/[^\\s/]+)+$",
```
Comment (sanderegg) [inline api/specs/webserver/v0/components/schemas/project-v0.0.1.json]: what about /demodec? probably could be removed...

**Pull request merged**